python/postprocessing/modules/btv/btagSFProducer.py

```diff
@@ -1,45 +1,45 @@
 import ROOT
 import os
+from itertools import chain
+
 ROOT.PyConfig.IgnoreCommandLineOptions = True
 
 from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection
 from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
 
-def is_relevant_syst_for_shape_corr(flavor_btv, syst):
+def is_relevant_syst_for_shape_corr(flavor_btv, syst, jesSystsForShape=["jes"]):
     """Returns true if a flavor/syst combination is relevant"""
+    jesSysts = list(chain(*[ ("up_" + j, "down_" + j) for j in jesSystsForShape ]))
+
     if flavor_btv == 0:
         return syst in [ "central",
-                         "up_jes", "down_jes",
                          "up_lf", "down_lf",
                          "up_hfstats1", "down_hfstats1",
-                         "up_hfstats2", "down_hfstats2" ]
+                         "up_hfstats2", "down_hfstats2" ] + jesSysts
     elif flavor_btv == 1:
         return syst in [ "central",
                          "up_cferr1", "down_cferr1",
                          "up_cferr2", "down_cferr2" ]
     elif flavor_btv == 2:
         return syst in [ "central",
-                         "up_jes", "down_jes",
                          "up_hf", "down_hf",
                          "up_lfstats1", "down_lfstats1",
-                         "up_lfstats2", "down_lfstats2" ]
+                         "up_lfstats2", "down_lfstats2" ] + jesSysts
     else:
         raise ValueError("ERROR: Undefined flavor = %i!!" % flavor_btv)
     return True
 
 class btagSFProducer(Module):
     """Calculate btagging scale factors
-        algo has to be either 'csvv2' or 'cmva'
     """
-    def __init__(self, era, algo = 'csvv2', sfFileName = None, verbose = 0, collName='Jet',storeOutput=True):
+    def __init__(self, era, algo='csvv2', selectedWPs=['M', 'shape_corr'], sfFileName=None, verbose=0, jesSystsForShape=["jes"]):
```

Comment:
Creo que este constructor se carga algo en cmgtools down the line, aunque no se si está en uso https://github.com/uniovi-hepex/cmgtools-lite/blob/7a7469722f740c181607f385ad865092da34ba57/TTHAnalysis/python/tools/nanoAOD/SMP_5TeV_modules.py#L436

Reply:
estos cambios son del central no mios